syntax = "proto3";

package spaceship;

service SpaceshipReporting {
  rpc GetSpaceships (Coordinates) returns (stream Spaceship) {}
}

message Coordinates {
  string coordinates = 1;
}

message Spaceship {
  enum Alignment {
    ALLY = 0;
    ENEMY = 1;
  }
  Alignment alignment = 1;
  string name = 2;
  string ship_class = 3;
  float length = 4;
  int32 crew_size = 5;
  bool armed = 6;
  repeated Officer officers = 7;
}

message Officer {
  string first_name = 1;
  string last_name = 2;
  string rank = 3;
}